convert/lib/nuke.py

```diff
@@ -0,0 +1,434 @@
+#!/usr/bin/env python3
+import boto3
+import re
+from pprint import pprint
+
+from .meta import cdk_ids
+
+
+class nuke:
+    def __init__(self, region: str, verbose: bool = False, delete: bool = False):
+        self.region = region
+        self.verbose = verbose
+        self.delete = delete
+
+    @property
+    def autoscaling(self):
+        if not getattr(self, "_autoscaling", None):
+            self._autoscaling = boto3.client("autoscaling", self.region)
+        return self._autoscaling
+
+    @property
+    def ec2(self):
+        if not getattr(self, "_ec2", None):
+            self._ec2 = boto3.client("ec2", self.region)
+        return self._ec2
+
+    @property
+    def iam(self):
+        if not getattr(self, "_iam", None):
+            self._iam = boto3.client("iam", self.region)
+        return self._iam
+
+    @property
+    def awslambda(self):
+        if not getattr(self, "_lambda", None):
+            self._awslambda = boto3.client("lambda", self.region)
+        return self._awslambda
+
+    @property
+    def ssm(self):
+        if not getattr(self, "_ssm", None):
+            self._ssm = boto3.client("ssm", self.region)
+        return self._ssm
+
+    @property
+    def stepfunctions(self):
+        if not getattr(self, "_stepfunctions", None):
+            self._stepfunctions = boto3.client("stepfunctions", self.region)
+        return self._stepfunctions
+
+    def _aws_list_all(
```

Comment:
Definitely see if the paginators help here?

Reply:
Yep, quite a bit cleaner now!